### PR TITLE
docs: add config-as-code docs

### LINKED ISSUE #14: Improvement: Document initial code-as-config setup
> [!IMPORTANT]
> Make sure to link this issue with the PR for your improvement.

GitHub organization is managed by otterdog, a config-as-code solution. There is already a minimal setup available which will be developed further. Initial documentation should be provided, that explains the setup and the configuration.

### PR DESCRIPTION
**Pull request by ltekieli**
Issue-ref: resolves #14

### CODE REVIEW

File: docs/features/integration/config_as_code.rst [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+Automatically generated documentation for eclipse-score otterdog
+configuration is here: https://eclipse-score.github.io/.eclipsefdn/
+
+All changes to the configuration should be handled by pull requests in
+https://github.com/eclipse-score/.eclipsefdn/
+
+Organization settings
+---------------------
+
+``default_branch_name`` - the default branch name for newly crated
```
Comment (masc2023) [inline docs/features/integration/config_as_code.rst:46]: crated -> created?

**Review by masc2023 — COMMENTED** [review]
Unsure that config as code is sub-feature of integration, is not Configuration and setting up repos an infrastructure topic?

**Comment by ltekieli** [comment]
> Unsure that config as code is sub-feature of integration, is not Configuration and setting up repos an infrastructure topic?

@masc2023 Would you move it as a separate feature?

File: docs/features/integration/config_as_code.rst (hunk 1 above)
Comment (ltekieli) [reply docs/features/integration/config_as_code.rst]: Fixed.

File: docs/features/integration/config_as_code.rst [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+Automatically generated documentation for eclipse-score otterdog
+configuration is here: https://eclipse-score.github.io/.eclipsefdn/
+
+All changes to the configuration should be handled by pull requests in
+https://github.com/eclipse-score/.eclipsefdn/
+
+Organization settings
+---------------------
+
+``default_branch_name`` - the default branch name for newly created
```
Comment (antonkri) [inline docs/features/integration/config_as_code.rst:46]: Could you make a bullet points out of it?

File: docs/features/integration/config_as_code.rst [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+repositories is set to ``main``.
+
+``has_discussions`` - organisation level discussions are enabled.
+``discussion_source_repository`` - organization level discussiones are
+attached to ``eclipse-score/score`` repository.
+
+Repository settings
+-------------------
+
+``homepage`` - the link that is displayed when opening the repository in
```
Comment (antonkri) [inline docs/features/integration/config_as_code.rst:56]: same here

File: docs/features/integration/config_as_code.rst (hunk 2 above)
Comment (ltekieli) [reply docs/features/integration/config_as_code.rst]: Fixed.

File: docs/features/integration/config_as_code.rst (hunk 3 above)
Comment (ltekieli) [reply docs/features/integration/config_as_code.rst]: Fixed.

**Review by antonkri — APPROVED** [review]
LGTM

**Pull request merged**